backend/src/Squidex/Areas/IdentityServer/Controllers/Account/AccountController.cs

```diff
@@ -240,6 +240,21 @@ public async Task<IActionResult> ExternalCallback(string? returnUrl = null)
         if (isLoggedIn)
         {
             user = await userService.FindByLoginAsync(login.LoginProvider, login.ProviderKey, HttpContext.RequestAborted);
+
+            if (user != null)
+            {
+                var squidexClaimsFromLoginPrincipal = login.Principal.Claims.GetSquidexClaims();
+
+                if (user.Claims.GetSquidexClaims().Equals(squidexClaimsFromLoginPrincipal))
```

Comment:
I think Equals does not work here. Must be something like SetEquals or so. It would also override all claims that have been set manually, not sure if this is intended.

Reply:
Ah good point, for our use case we absolutely want permissions in Squidex to exactly match what is in our providers configuration but perhaps others may not want the same behaviour; could break existing user's setups if they have only manually setup permissions and just use the external provider to login. Does this mean it'd need a new configuration option called something like `oidcSyncPermissionsWithClaimsOnLogin` that's false by default?

Thanks for the tip around `SetEquals`, knew I was optimistic just putting `Equals` in there! Will try and actually test it this time before I push it up for review.

Reply:
Yes, another option would be good.

`GetSquidexClaims` does return `IEnumerable<Claim>` which does not implement Equals, therefore you have to find another method for that. Also better call ToList() if you make the call twice to the same user.

Reply:
Added configuration option but ran out of time today (and not working tomorrow) so simply removed my invalid claims comparison for now to see what you think about relying on your existing `SyncClaims` code to only update what is needed which is tried and tested? Seems like it should be pretty quick to go through `UpdateAsync` and then `SyncClaims`, and only happens on login for people making use of this new setting, however I can understand how any performance overhead should be avoided if possible.